dataset-processor-python/data_loader.py

```diff
@@ -1,55 +1,272 @@
+import json
 import os
+import re
 import sys
 import csv
+import pandas as pd
+import numpy
 
 
-class DataLoader:
+class UserInputHandler:
+    """
+    Class to handle user inputs
+    """
+
     def __init__(self, path, dataset_name):
         self.path = path
         self.dataset_name = dataset_name
-        self.initial_data = None
+        self.inputs = self.get_user_inputs()
+
+    @staticmethod
+    def is_valid_version(version):
+        pattern = r"^\d{4}\.[0-9]+$"
```

Comment:
Nit: `\d` and `[0-9]` are equivalent, probably better to pick one and stick with it? I personally prefer `[0-9]` because it's a little more readable without knowing specific regex syntax but anything is fine.